testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/preload/PreloadMiniAppWindow.kt

```diff
@@ -155,15 +129,48 @@ class PreloadMiniAppWindow(
         }
     }
 
-    private fun isAccepted(): Boolean {
-        try {
-            if (prefs.contains(miniAppId)) return prefs.getBoolean(miniAppId, DEFAULT_ACCEPTANCE)
-        } catch (e: Exception) {
-            return false
+    private fun isAcceptedRequiredPermissions(manifest: MiniAppManifest): Boolean {
```

Comment:
The SDK throws an `Exception` when the required permissions aren't granted, so couldn't you just catch that `Exception` and display the first-time launch screen? (I think this is what iOS is doing) I'm not sure you need to manually check the required permissions here.

By the way, is there a specific `Exception` type for this case where required permissions aren't granted? If not, I think we should add one (similar to iOS) like `RequiredPermissionsNotGrantedException`.

Reply:
> The SDK throws an Exception when the required permissions aren't granted, so couldn't you just catch that Exception and display the first-time launch screen? (I think this is what iOS is doing) I'm not sure you need to manually check the required permissions here.

Yeah, I was about to start like this, but I realized there is something refactoring I need to figure out, so I have check it manually without catching the exception.

> By the way, is there a specific Exception type for this case where required permissions aren't granted? If not, I think we should add one (similar to iOS) like RequiredPermissionsNotGrantedException.

Updated.